v2/tools/generator/internal/codegen/code_generator.go

```diff
@@ -137,6 +137,7 @@ func createAllPipelineStages(idFactory astmodel.IdentifierFactory, configuration
 		// TODO: These should be removed if/when we move to Swagger as the single source of truth
 		pipeline.RemoveTypeProperty(),
 		pipeline.RemoveAPIVersionProperty(),
+		pipeline.AddAPIVersionEnums(),
```

Comment:
Are these two lines removing something and then adding it right back again? 
If so, does this need to happen further upstream?

Reply:
I think the distinction is between the property (on the resource) containing an APIVersion field, and the actual API versions themselves (in the package, but not as a field on the resource)

Reply:
Precisely.